ingestion/src/metadata/ingestion/source/database/snowflake/lineage.py

```diff
@@ -61,3 +66,33 @@ def get_stored_procedure_queries_dict(self) -> Dict[str, List[QueryByProcedure]]
         )
 
         return queries_dict
+
+    def yield_table_query(self) -> Iterator[TableQuery]:
+        """
+        Given an engine, iterate over the query results to
+        yield a TableQuery with query parsing info
+        """
+        for engine in self.get_engine():
+            rows = None
+            with engine.connect() as conn:
+                rows = conn.execute(
+                    self.get_sql_statement(
+                        start_time=self.start,
+                        end_time=self.end,
+                    )
+                )
```

Comment:
do we not need to do fetch all?

Reply:
yes it would be better since we are exiting the connection but not sure if count of rows is too large then it would cause issue or not !